### PR TITLE
Add release guide to the website

### PR DESCRIPTION
**Pull request by dhalperi**
R: @davorbonaci

### CODE REVIEW

File: contribute/release-guide.md [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+
+**********
+
+## Decide to release
+
+Deciding to release and selecting a Release Manager is the first step of the release process. This is a consensus-based decision of the entire community.
+
+Anybody can propose a release on the dev@ mailing list, giving a solid argument and nominating a committer as the Release Manager (including themselves). There’s no formal process, no vote requirements, and no timing requirements. Any objections should be resolved by consensus before starting the release.
+
+In general, the community prefers to have a rotating set of 3-5 Release Managers. Keeping a small core set of managers allows enough people to build expertise in this area and improve processes over time, without Release Managers needing to re-learn the processes for each release. That said, if you are a committer interested in serving the community in this way, please reach out to us on the dev@ mailing list.
```
Comment (nssalian) [inline contribute/release-guide.md]: would suggest to have this: "please reach out to us on the dev@ mailing list."
To "please reach out to the community on the dev@ mailing list" unless there is a specific list for committers / PMC members

File: contribute/release-guide.md [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+          </servers>
+        </settings>
+
+#### Website development setup
+
+Get ready for updating the Beam website by following the [website development instructions](/contribute/contribution-guide/#website).
+
+### Create a new version in JIRA
+
+When contributors resolve an issue in JIRA, they are tagging it with a release that will contain their changes. With the release currently underway, new issues should be resolved against a subsequent future release. Therefore, you should create a release item for this subsequent release, as follows:
```
Comment (nssalian) [inline contribute/release-guide.md]: I was under the impression that committers who resolve the issues determine the suitable release for the fix. Please correct me if I am wrong. Generally seen that contributors push code to the master and then they get added to branches. If separation is need a new PR/ patch would be done specifically for that branch.

**Comment by pabloem** [comment]
r: @robertwb please review : )

File: contribute/release-guide.md (hunk 1 above)
Comment (dhalperi) [inline contribute/release-guide.md]: Will fix, thanks.

File: contribute/release-guide.md (hunk 2 above)
Comment (dhalperi) [inline contribute/release-guide.md]: Typically fixes will go into the development version, which is the next minor release. We additionally will cherry-pick from master into release branches, but only for hot fixes. What you are proposing is good, and a more flexible process, but we are just not there yet. For now I'll keep as-is but this is something we should consider doing in the near future.

**Pull request closed without merging**